src/core/search/indices.h

```diff
@@ -39,25 +45,39 @@ struct BaseStringIndex : public BaseIndex {
   const CompressedSortedSet* Matching(std::string_view str) const;
 
  protected:
-  absl::flat_hash_map<std::string, CompressedSortedSet> entries_;
+  CompressedSortedSet* GetOrCreate(std::string_view word);
+
+  template <typename K, typename V, typename H, typename E, typename A>
+  static auto WithPmr(absl::flat_hash_map<K, V, H, E, A> m)
+      -> absl::flat_hash_map<K, V, H, E, std::pmr::polymorphic_allocator<std::pair<const K, V>>> {
+    return {nullptr};
+  }
+
+  decltype(WithPmr(absl::flat_hash_map<std::string, CompressedSortedSet>{})) entries_;
 };
```

Comment:
I _tried_ accessing a contructed hash maps hasher and key_equality with member types, but `hasher` and `key_equal` just don't work, its hidden inside the internal policy 😵 

The other option is using internal absl types to pass the default hasher/key_equal.

I prefer this function. It can also be done with a class to get rid of the decltype below... But I hope this does it